### PR TITLE
CHORE Remove unused files and deprecation warnings

### PR DESCRIPTION
**Pull request by bsaris-ut**
Did a lot of this work while digging into spec failures and didn't want to lose it so have this follow up PR.

Update Rakefile, database cleaner gem, and parser_spec to remove warnings.

Remove locales files and reference to empty directories in generators.

Removed some should matchers in specs.

### CODE REVIEW

File: surveyor.gemspec [hunk 1]
```diff
@@ -31,7 +31,7 @@ Gem::Specification.new do |s|
   s.add_development_dependency('poltergeist', '~>1.17.0')
   s.add_development_dependency('json_spec', '~> 1.1.1')
   s.add_development_dependency('factory_bot', '~> 4.8.0')
-  s.add_development_dependency('database_cleaner', '~> 1.2.0')
+  s.add_development_dependency('database_cleaner', '~> 1.2')
```
Comment (jeffbax) [inline surveyor.gemspec:34]: I think a bunch of these seem to not match the rails app

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -33,8 +32,6 @@
   config.include SurveyorAPIHelpers
   config.include WaitForAjax
 
-  config.treat_symbols_as_metadata_keys_with_true_values = true
-
```
Comment (jeffbax) [inline spec/spec_helper.rb:37]: I thought this was added, and isn't this true in the rails app?

**Review by jeffbax — COMMENTED** [review]
are we removing support for translations? is it not useful to have that stuff even if "unused" ?

File: spec/models/survey_section_spec.rb [hunk 3]
```diff
@@ -261,16 +265,16 @@
...
 
       new_set.responses.build(
         :question_id => dependency_mandatory_question.id,
         :answer_id => dependency_mandatory_answer.id,
         :string_value => "answer"
       )
-      new_set.save.should be true
-      dependency_section.completed?( new_set ).should be true
+      new_set.save!
+      expect(dependency_section.completed?(new_set)).to be(true)
```
Comment (AlexAlt) [inline spec/models/survey_section_spec.rb:277]: how did these get missed? thanks for catching it

File: spec/spec_helper.rb (hunk 2 above)
Comment (AlexAlt) [reply spec/spec_helper.rb:37]: yeah I think we want to keep this for database cleaner

File: surveyor.gemspec (hunk 1 above)
Comment (bsaris-ut) [reply surveyor.gemspec:34]: Probably -- but we don't use this gem in our app (this is only a dev dependency) & was causing a warning.

File: spec/spec_helper.rb (hunk 2 above)
Comment (bsaris-ut) [reply spec/spec_helper.rb:37]: The warnings I got said that this was defaulted to true now and so this config wasn't necessary any more.

File: spec/models/survey_section_spec.rb (hunk 3 above)
Comment (bsaris-ut) [reply spec/models/survey_section_spec.rb:277]: There are a few more -- these were just in the files I was playing with when trying to figure out what was going on. I don't think this actually breaks rspec 3, it just gets sad. We likely should go through and get all of them but this was mostly just clean up that we didn't _need_ but I didn't want to lose.

Comment (jeffbax) [reply spec/models/survey_section_spec.rb:277]: I thought my branch did all of this  originally.... there's a 2->3 upgrade tool

**Review by jeffbax — APPROVED** [review]

**Review by AlexAlt — APPROVED** [review]
LGTM!

**Comment by bsaris-ut** [comment]
The translations we're removing are around their FE implementation so it doesn't really apply to us.

**Pull request merged**